README.md

```diff
@@ -1015,6 +1015,7 @@ The following constructs are either unsupported or almost never work:
 * rank-n types
 * promoting `TypeRep`s
 * `TypeApplications`
+* `{-# UNPACK #-}` pragmas
```

Comment:
Can't we just drop these pragmas? That seems strictly more expressive, allowing users to unpack their unrefined datatypes, even if we don't do this for the singletons we generate.

EDIT: Actually, it seems this is what we do, from reading the code. But then this README is misleading.

Reply:
The treatment of `UNPACK` patterns here is basically identical to how we treat bang patterns during promotion—we drop them and emit a warning saying as much. (Granted, the fact that we drop bang patterns during promotion is not documented in the `README`. It should be.)

How would you phrase this to be less misleading?

Reply:
It's this text: 

```
Attempting to single a data constructor with an `{-# UNPACK #-}` pragma
(e.g., `data T = MkT {-# UNPACK #-} !()`) will not work.
```

Maybe just say the `{-# UNPACK #-}` will be ignored.

Reply:
Ah. I think I intended that to say "won't work as you might expect", but it ended up sounding more uncompromising than I had intended. I've reworded this to use your suggesting phrasing, and I've adopted similar phrasing in the newly minted sections on bang patterns and irrefutable patterns.